src/main/java/core/basesyntax/SupplyOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
```

Comment:
The implementation of `handle` method should not directly interact with the storage map. It's better to use a service layer for such interactions to maintain a clean separation of concerns and follow the Single Responsibility Principle.